### PR TITLE
Fix DefaultConnectionPlugin inTransaction bug

### PR DESCRIPTION
**Pull request by aaron-congo**
- Fixed a bug where methods invoked against old connections changed the current connection inTransaction status

### CODE REVIEW

**Review by joyc-bq — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java [hunk 1]
```diff
@@ -99,6 +100,11 @@ public <T, E extends Exception> T execute(
     final T result = jdbcMethodFunc.call();
 
     Connection currentConn = this.pluginService.getCurrentConnection();
+    final Connection methodInvokeOnConn = WrapperUtils.getConnectionFromSqlObject(methodInvokeOn);
```
Comment (sergiyvamz) [inline wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java:103]: I'd rename it to something like "boundedConnection"

Comment (sergiyvamz) [reply wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java:103]: Do we need to check if method is close() or abort() and skip this check?

Comment (aaron-congo) [reply wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java:103]: I don't think so, the method is still executed above on line 100 regardless of if we are using an old connection or not. This check just makes sure we don't set the inTransaction or autocommit value for the current connection if the one passed into this method is old

**Review by karenc-bq — APPROVED** [review]

**Review by joyc-bq — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java (hunk 1 above)
Comment (sergiyvamz) [reply wrapper/src/main/java/software/amazon/jdbc/plugin/DefaultConnectionPlugin.java]: Good point. Then I suggest to move this code before line 100 and add check for close/abort methods.

**Review by sergiyvamz — APPROVED** [review]

**Pull request merged**